Formula/g/gnome-online-accounts.rb

```diff
@@ -0,0 +1,122 @@
+class GnomeOnlineAccounts < Formula
+  desc "Single sign-on framework for GNOME"
+  homepage "https://gitlab.gnome.org/GNOME/gnome-online-accounts"
+  url "https://download.gnome.org/sources/gnome-online-accounts/3.52/gnome-online-accounts-3.52.3.1.tar.xz"
+  sha256 "49ed727d6fc49474996fa7edf0919b21e4fc856ea37e6e30f17b50b103af9701"
+  license "LGPL-2.0-or-later"
+  head "https://gitlab.gnome.org/GNOME/gnome-online-accounts.git", branch: "master"
+
+  depends_on "dbus" => :build
+  depends_on "docbook-xsl" => :build
+  depends_on "gobject-introspection" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkgconf" => [:build, :test]
+  depends_on "vala" => :build
+
+  depends_on "glib"
+  depends_on "gtk4"
+  depends_on "json-glib"
+  depends_on "libadwaita"
+  depends_on "libgoa"
+  depends_on "librest"
+  depends_on "libsecret"
+  depends_on "libsoup"
+
+  uses_from_macos "libxslt" => :build # for xsltproc
+  uses_from_macos "libxml2"
+
+  on_macos do
+    depends_on "gettext"
+  end
+
+  on_linux do
+    depends_on "gcr"
+    depends_on "keyutils"
+    depends_on "krb5"
+  end
+
+  def install
+    # NOTE: `libgoa` is split out to provide a lightweight formula with minimal dependencies
+    inreplace "src/meson.build", /^subdir\('goa'\)$/, "libgoa_dep = dependency(goa_api_name)"
+    rm_r("src/goa")
```

Comment:
Trying out a split formula here so resulting distribution is similar to Arch (https://archlinux.org/packages/extra/x86_64/libgoa/). This requires some more workarounds in Homebrew as we don't natively support splitting packages.

Mainly want this for reasons similar to our other split formulae where the client library is more commonly used dependency with minimal dependencies (`glib`) while full featured backend will result in having a GTK dependency which is less ideal as this results in multiple GTK versions in dependency trees.